### PR TITLE
chore: use react 17

### PR DESCRIPTION
**Pull request by richbachman**
## Description of the change

I'm been working on adding Pluto to the Backend repo which runs React 17. I'm not able to use any components because they're built in React 18. This changes the deps to use 17. They components will still work in React 18 proejcts.

## Testing the change

- [ ] Write your testing instructions here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.

### CODE REVIEW

**Review by RohitRajendran — APPROVED** [review]
Lgtm!

**Comment by richbachman** [comment]
@RohitRajendran can I get another set of 👀 on this one?

**Review by RohitRajendran — APPROVED** [review]

File: packages/components/src/components/Callout/Callout.test.tsx [hunk 1]
```diff
@@ -9,12 +9,6 @@ describe("<Callout />", () => {
     expect(screen.getByText("Message here")).toBeInTheDocument();
   });
 
-  it("should render the icon", () => {
-    render(<Callout>Message here</Callout>);
-
-    expect(screen.getByTestId("callout-icon")).toBeInTheDocument();
-  });
```
Comment (apvale) [inline packages/components/src/components/Callout/Callout.test.tsx:16]: Is this covered by Chromatic, right?

Comment (apvale) [reply packages/components/src/components/Callout/Callout.test.tsx:16]: Btw, why did you remove the test-id and this test? It is something related to the downgrade?

Comment (richbachman) [reply packages/components/src/components/Callout/Callout.test.tsx:16]: Yeah. The icon is a visual treatment so will be caught there if its missing.

Comment (richbachman) [reply packages/components/src/components/Callout/Callout.test.tsx:16]: > Btw, why did you remove the test-id and this test? It is something related to the downgrade?

We shouldn't be forcing test-ids in our components. That can cause issues with teams using the components and their tests, i.e. they use the same test-id.

**Pull request merged**